android-core/src/main/java/com/mparticle/internal/KitFrameworkWrapper.java

```diff
@@ -77,12 +77,13 @@ public void loadKitLibrary() {
                             .onKitsLoaded(() -> {
                                         mKitManager = kitManager;
                                         setKitsLoaded(true);
+                                        updateDataplan(mCoreCallbacks.getDataplanOptions());
```

Comment:
this reads like a race condition - we need to ensure that the data plan is set before kits are loaded. specifically - we need to make sure that no user attribute, events, etc can be sent to kits prior to the data plan.

in my view ideally we'd load the data plan in the same function `updateKits` above that loads kit configuration. the idea of KitConfiguration is that it should contain absolutely everything needed, and we only start kits after it's been completely parsed.